edgelet/hsm-sys/azure-iot-hsm-c/src/edge_hsm_client_store.c

```diff
@@ -2103,6 +2103,91 @@ static int edge_hsm_client_store_remove_key
     return result;
 }
 
+static KEY_HANDLE open_key
+(
+    CRYPTO_STORE *store,
+    HSM_KEY_T key_type,
+    const char* key_name
+)
+{
+    KEY_HANDLE result;
+    bool do_key_create = true;
+
+    if (key_type == HSM_KEY_ENCRYPTION)
```

Comment:
Do these changes have any interaction with the update Raj just made?

Reply:
Just want to make sure we don't regress.

Reply:
Glad you noticed and for bringing this up. This part of the code is correct and stays the same as before. The bug in the other case was in function edge_hsm_client_store_insert_encryption_key. Here is the PR for the fix: https://github.com/Azure/iotedge/pull/1086.

This refactoring was merely to facilitate readability and not have long implementation in a function.